hadoop-plugin/src/main/groovy/com/linkedin/gradle/dependency/DependencyPlugin.groovy

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright 2015 LinkedIn Corp.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.linkedin.gradle.dependency;
+
+import org.gradle.api.Plugin;
+import org.gradle.api.Project;
+import org.gradle.api.Task;
+
+class DependencyPlugin implements Plugin<Project> {
+
+  // by default disable dependency check. Subclasses can set this variable to true to enable dependencyCheck
+  boolean enableDependencyCheck = false;
+
+  @Override
+  void apply(Project project) {
+    if (enableDependencyCheck) {
```

Comment:
I think we should not allow users to disable the dependencyPlugin by using -PdisableDependencyPlugin.  If we allow this, then users can always bypass the dependency check by disabling this plugin.